src/pytorch_lightning/core/saving.py

```diff
@@ -57,12 +56,12 @@ class ModelIO:
     @classmethod
     def load_from_checkpoint(
         cls,
-        checkpoint_path: Union[str, IO],
-        map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
+        checkpoint_path: _PATH,
+        map_location: MAP_LOCATION = None,
```

Comment:
```suggestion
        map_location: Optional[MAP_LOCATION] = None,
```

Reply:
reverted to https://github.com/Lightning-AI/lightning/pull/13932/commits/0502a4e9fe2f1b229265de8ae82a129e704f65ee and synced with upstream in https://github.com/Lightning-AI/lightning/pull/13932/commits/82fc4528431fd3412e30782ba64bcc65d2adba38